ocean_lib/ocean/test/test_ocean_assets.py

```diff
@@ -39,43 +40,56 @@ def test_register_asset(publisher_ocean_instance, publisher_wallet, consumer_wal
     assert ocn.assets.resolve(invalid_did) is None
 
 
-@pytest.mark.skip(reason="TODO: update function on OceanAssets class")
-def test_update(publisher_ocean_instance, publisher_wallet, consumer_wallet, config):
-    ocn = publisher_ocean_instance
-    block = ocn.web3.eth.block_number
-    alice = publisher_wallet
-    bob = consumer_wallet
+def test_update(publisher_ocean_instance, publisher_wallet, config, metadata=None):
+    """Helper function for asset update based on ddo_sa_sample.json."""
+    ocean = publisher_ocean_instance
+    erc20_data = ErcCreateData(
+        template_index=1,
+        strings=["Datatoken 1", "DT1"],
+        addresses=[
+            publisher_wallet.address,
+            publisher_wallet.address,
+            ZERO_ADDRESS,
+            get_address_of_type(config, "Ocean"),
+        ],
+        uints=[ocean.web3.toWei("0.5", "ether"), 0],
+        bytess=[b""],
+    )
 
-    ddo = create_asset(ocn, alice, config)
+    if not metadata:
```

Comment:
well, we are testing the update. When updating metadata, metadata should be required, so the option of metadata being empty/None is moot.